packages/js/core/src/types/WrapError.ts

```diff
@@ -0,0 +1,274 @@
+import { CleanResolutionStep } from "../algorithms";
+
+export type ErrorSource = Readonly<{
+  file?: string;
+  row?: number;
+  col?: number;
+}>;
+
+/**
+Wrap error codes provide additional context to WrapErrors.
+
+Error code naming convention (approximate):
+    type of handler
+    type of functionality
+    piece of functionality
+    ==> handler_typeFn_pieceFn
+
+Error code map:
+  0-24 -> Client
+  25-49 -> URI resolution
+  50-74 -> Wrapper invocation & sub-invocation
+  75-255 -> Unallocated
+ */
+export enum WrapErrorCode {
+  UNKNOWN,
+  CLIENT_LOAD_WRAPPER_ERROR,
+  CLIENT_GET_FILE_ERROR,
+  CLIENT_GET_IMPLEMENTATIONS_ERROR,
+  CLIENT_VALIDATE_RESOLUTION_FAIL,
+  CLIENT_VALIDATE_ABI_FAIL,
+  CLIENT_VALIDATE_RECURSIVE_FAIL,
+  URI_RESOLUTION_ERROR = 25,
+  URI_RESOLVER_ERROR,
+  URI_NOT_FOUND,
+  WRAPPER_INVOKE_ABORTED = 50,
+  WRAPPER_SUBINVOKE_ABORTED,
+  WRAPPER_INVOKE_FAIL,
+  WRAPPER_READ_FAIL,
+  WRAPPER_INTERNAL_ERROR,
+  WRAPPER_METHOD_NOT_FOUND,
+  WRAPPER_ARGS_MALFORMED,
+}
+
+export interface WrapErrorOptions {
+  code: WrapErrorCode;
+  uri: string;
+  method?: string;
+  args?: string;
+  source?: ErrorSource;
+  resolutionStack?: CleanResolutionStep;
+  cause?: unknown;
+  prev?: Error;
```

Comment:
Isn't `prev` always a WrapError?
Also, can we get a more descriptive name for it, maybe `innerError` or something else?

Reply:
Yes, you're right!